VAMobile/documentation/docs/Engineering/FrontEnd/DevelopmentBestPractices.md

```diff
@@ -0,0 +1,135 @@
+
+Use a feature-xl branch when you are working on a large feature that requires multiple sprints and tickets to complete. It contains all your work prior to release of the feature.
+
+:::info
+For features that must be rolled out gradually, or where precise feature activation timing is required, use a feature flag instead of a feature-xl branch.
+:::
+
+Before you start work, create a branch with the "feature-xl" prefix like:
+
+```
+feature-xl/my-big-new-feature
+```
+
+Then create a smaller branch off the feature-xl branch for each ticket. When all the work is complete and the smaller branches are merged back into the feature-xl branch, QA will perform a final check on the feature-xl branch. Then you can merge the feature-xl branch back into the develop branch.
+
+## Pull Requests
+
+A Pull Request (PR) is a request to merge code. All pull requests must include certain information. Each PR must pass a series of manual and automated checks before it can be merged.
+
+### Pull request process
+
+Pull requests move through a series of steps.
+
+1. Creation: an engineer creates a PR. See "Pull request creation" below.
+2. Review: a different engineer reviews the PR. See "Pull request review" below.
+3. Approval: when the reviewer is satisfied, they approve the PR and change the pipeline to "With QA (pre-develop)" so QA knows to begin testing.
+4. QA: a QA engineer tests the branch and updates TestRail. QA informs the PR creator of any issues they find. On approval, QA comments in the ticket and changes the pipeline to "Ready to merge to develop".
+5. Merge: PR creator hits "Merge pull request" to merge their branch into the develop branch. On the ticket, PR creator changes the pipeline to "With QA (develop)" and selects the current release under "Releases".
+6. Post-merge: QA completes final testing and closes the ticket.
+
+### Pull request creation
+
+When creating a PR, fill out each section in the PR template, including:
+- Description: provide context so a code reviewer or future developer understands why the PR exists and what it does
+- Screenshots: add before/after screenshots or videos showing the changes
+- Testing information: confirm testing on iOS and Android. Document any special considerations like edge cases or areas of focus for QA
+- Reviewer validations: tell reviewers what to check. You can usually copy the Acceptance Criteria from the ticket
+- Complete all PR checklist items
+
+Immediately after creating the PR, click the "Connect issue" button and choose the associated ticket. You must be logged into ZenHub to see the "Connect issue" button.
+
+:::important
+It's critical to link the PR to the ticket with the "Connect" issue" button so we can track the work performed.
+:::
+
+Add the "FE-Needs Review" label. This indicates to other engineers that your PR is ready for review. If your PR is especially complex, you can annotate the PR with comments to help reviewers understand why you made particular decisions.
```

Comment:
Do we need these labels? I feel like anything that isn't in "draft" needs FE review and there's a "code review" zenhub lane -- same comment for the QA label, we have a "with QA" zenhub lane.

Reply:
I think we added the labels because we had a lot of PRs and everyone was on one big team, so it was hard to keep track of everything. We can revisit now that we have smaller teams. One note, the labels are automated except for the initial "FE-Needs Review" label. I've removed mention of that label from the docs here